appservice/appservice.go

```diff
@@ -271,7 +301,7 @@ func (as *AppService) makeClient(userID id.UserID) *mautrix.Client {
 	client.StateStore = as.StateStore
 	client.SetAppServiceUserID = true
 	client.Log = as.Log.With().Str("as_user_id", client.UserID.String()).Logger()
-	client.Client = as.HTTPClient
+	client.Client = as.CreateHTTPClient()
```

Comment:
Creating new clients for each ghost user doesn't seem like a good idea 🤔

Reply:
Right, but now it is possible to have a mix of app servers with different connection (tcp vs unix). Should I cache http clients per type and lazy create them?

Reply:
At the same time, are you worried about memory/performance of creating http.client?

Reply:
The `AppService` struct has the http client as a field so it's not shared between different instances or anything like that.

Having multiple http clients means connections don't get reused properly

Reply:
ok, looked at again, my bad, fixed it, I was confused by the fact that AppService is created and only then also populated by config values from yaml, I think this is super confusing.

Reply:
@tulir is this better? I am testing it with dendrite and whatsapp bridge, but I think they do not work together well currently because of this: https://github.com/matrix-org/dendrite/issues/2902 other than this I think two-way unix socket communication is working ok.

Reply:
@tulir is it possible to make a decision on this PR?